spec/integration/autobahn_intg_spec.rb

```diff
@@ -319,6 +319,20 @@ def create_transport_system(name, options={})
         await_delivery
         @queues.map { |q| q.status.first }.reduce(:+).should == 1
       end
+
+      it 'does not deliver any new messages after it has been unsubscribed' do
+        unsubscribed = Autobahn::Concurrency::Semaphore.new(0)
+        @consumer.subscribe do |headers, message|
+          begin
+            unsubscribed.acquire
+            headers.ack
+          rescue java.lang.InterruptedException
+          end
+        end
+        unsubscribed.release
+        @consumer.unsubscribe!
```

Comment:
The test is not to ensure whether `#ack` is called either before or after `#unsubscribe!`, as what I wanted to test was that there wouldn't be any more calls to the given `handler` after `#unsubscribe!` returned.